### ISSUE TITLE
"until" field throws exception if given output of Chrome's Date.prototype.toISOString due to milliseconds

### ISSUE THREAD

**Issue by jlennox** [jakubroztocil/rrule#77]
The issue is the regex match does not account for the milliseconds represented as decimals, ie in "2014-11-18T02:02:07.443Z" the ".443" is unexpected.

Chrome 38.0.2125.111 m
Verified on latest rrule.js

**Comment by jsilvestre** [comment]
I'm going to answer one year later, because I was about to fill a bug which is not one, and it made me read the RFC carefully.

**TL;DR:** `.toISOString` returns something which is not what the iCal standard expects. "20141118T020207Z" is the valid version of the date you want.

**Long version, backed by the RFC:**
The standard says that UNTIL is a type "enddate", which means it can be a "date" or a "date-time". See https://tools.ietf.org/html/rfc5545#section-3.3.10 for source.

* "date" is quite obvious, and is not the point of your original issue so I skip it.

* "date-time" is basically "date "T" time", you can browse the examples in this section: https://tools.ietf.org/html/rfc5545#section-3.3.5

It means its value can be:
* 19980118T230000
* 19980118T230000Z
* TZID=America/New_York:19980119T020000

If we read further the description of the RRULE field ([here](https://tools.ietf.org/html/rfc5545#section-3.3.10)), we will find more information / contraints to apply to the field (I split the rules to ease reading).

> [...]The value of the UNTIL rule part MUST have the same value type as the "DTSTART" property. 

So it must be "date", or "date-time", but the same as DTSTART.

> Furthermore, if the "DTSTART" property is specified as a date with local time, then the UNTIL rule part MUST also be specified as a date with local time.

> If the "DTSTART" property is specified as a date with UTC time or a date with local time and time zone reference, then the UNTIL rule part MUST be specified as a date with UTC time.

> In the case of the "STANDARD" and "DAYLIGHT" sub-components the UNTIL rule part MUST always be specified as a date with UTC time.

> If specified as a DATE-TIME value, then it MUST be specified in a UTC time format.

This explains why you can't have "TZID=America/New_York:19980119T020000", or even "19980119T020000".

> If not present, and the COUNT rule part is also not present, the "RRULE" is considered to repeat forever.
